### ISSUE TITLE
Version8: Problem with Gulp and Windows paths

### ISSUE THREAD

**Issue by hardreddata** [TerriaJS/terriajs#4563]
I thought I would take a look at a clean clone of the _next_ branch which I cloned to `D:\dev\TerriaNext` on my Windows 10 PC.

The `npm install` worked. Shown below is an extract of the Gulp output. Notice the `Module not found: Error: Can't resolve 'D:devTerriaNext`, and the following `\newline` and `\tab` from **N**ode and **T**erria. 

I am using Git Bash.

```
...
 f5ad078d6d4ff7fc08ef6a3105dc2ef9.jpg   24.9 KiB                 [emitted]
f65ca6d17be7dd1611465960983a262b.json     64 KiB                 [emitted]
facde137ccef1e22b31b75bb0db0c985.json   64.3 KiB                 [emitted]
                  index.TerriaMap.css   48.2 KiB          index  [emitted]  index
                   index.TerriaMap.js    110 KiB          index  [emitted]  index
          vendors~index.TerriaMap.css    327 KiB  vendors~index  [emitted]  vendors~index
           vendors~index.TerriaMap.js   43.7 MiB  vendors~index  [emitted]  vendors~index
Entrypoint main = TerriaMap.js

ERROR in ./node_modules/terriajs/lib/Map/CesiumTileLayer.ts
Module not found: Error: Can't resolve 'D:devTerriaNext
ode_modules     erriajslib/Core/superGet' in 'D:\dev\TerriaNext\node_modules\terriajs\lib\Map'
 @ ./node_modules/terriajs/lib/Map/CesiumTileLayer.ts 104:11-79
 @ ./node_modules/terriajs/lib/Models/Leaflet.ts
 @ ./node_modules/terriajs/lib/ViewModels/TerriaViewer.ts
 @ ./node_modules/terriajs/lib/Models/Terria.ts
 @ ./index.js
 @ ./entry.js

ERROR in ./node_modules/terriajs/lib/ModelMixins/Cesium3dTilesMixin.ts
Module not found: Error: Can't resolve 'D:devTerriaNext
ode_modules     erriajslib/Core/superGet' in 'D:\dev\TerriaNext\node_modules\terriajs\lib\ModelMixins'
 @ ./node_modules/terriajs/lib/ModelMixins/Cesium3dTilesMixin.ts 154:11-79
 @ ./node_modules/terriajs/lib/Models/Cesium3DTilesCatalogItem.ts
 @ ./node_modules/terriajs/lib/Models/GlobeOrMap.ts
 @ ./node_modules/terriajs/lib/Models/NoViewer.ts
 @ ./node_modules/terriajs/lib/ViewModels/TerriaViewer.ts
 @ ./node_modules/terriajs/lib/Models/Terria.ts
 @ ./index.js
 @ ./entry.js

ERROR in ./node_modules/terriajs/lib/ModelMixins/CatalogMemberMixin.ts
Module not found: Error: Can't resolve 'D:devTerriaNext
ode_modules     erriajslib/Core/superGet' in 'D:\dev\TerriaNext\node_modules\terriajs\lib\ModelMixins'
...
```
It goes on to list a bunch of Mixins.

No rush.

Thanks!

**Comment by zoran995** [comment]
This is probably related to changes introduced in #4552. Quick fix (more hack) for this so you can work is to update [line 99 of configureWebpack](https://github.com/TerriaJS/terriajs/blob/next/buildprocess/configureWebpack.js#L99), to replace backslashes (`\`) with slashes (`/`)
something like this
```
path.resolve(terriaJSBasePath, 'lib').replace(/\\/g, "/")
```
Local build of application was successful after changing that line. Haven't checked if this is breaking build on other systems.

**Comment by zoran995** [comment]
Probably a better way is to escape the backslashes. Seems to be related to https://github.com/babel/babel/issues/9765

**Comment by hardreddata** [comment]
Thanks @zoran995 ; your _keep me moving_ solution worked great.

**Comment by stevenjh** [comment]
For future people, if this fix isn't working, double check your terriaMap package version matches that in your packages folder so updates on the files are being used in the build.  ie. Make sure the terriajs folder in node_modules is a link to the packages version

**Comment by hardreddata** [comment]
If you are short on windows PCs and want me to test anything please do let me know.

**Comment by tephenavies** [comment]
Still coming to this one. I've made a fix for the node v12.17.0+ bug #4617 on the way because it's really annoying to change node versions on `git bash` and it's sad to have 1 bug break our compatibility with the latest LTS node.

**Comment by tephenavies** [comment]
#4617 is fully fixed now so I'm on to this issue. I found that:
```
replace: 'var _get = require(String.raw`' + path.resolve(terriaJSBasePath, 'lib') + '/Core/superGet`).default;',
```
works well for Windows paths.

Now I have a new error building version 8 on Windows (under Git Bash):
```
ERROR in packages/terriajs/tsconfig.json
TS5083: Cannot read file 'C:\Users\Steve\Documents\Data61\TerriaMap\packages\terriajs\tsconfig.json'.
```
That file exists, but the build fails due to this error.

**Comment by zoran995** [comment]
@steve9164 just done some testing with above change and everything works fine on my end (yarn `build` and `watch` on TerriaMap, npm `install`, `build` and `watch` on TerriaJS). I am using git bash from VSC and there is no error. Node v12.13.1

Can do some more testing tomorrow if you need?

**Comment by hardreddata** [comment]
Working fine here. With the understanding (see gitter) that I had to bin `package-lock.json`.

I made the edit in place to line 99 of `configureWebpack.json` inside `node_modules\terriajs\buildprocess` after the `npm install`, and then ran gulp.

Node v10.15.0

I don't have a *packages* folder. Maybe this is some yarn stuff. Using npm here.

I hope this helps.

**Comment by nf-s** [comment]
With @zoran995 webpack config fix, everything is working for me so far natively on Windows - with Node v12.18.3 - using yarn workspaces.
It is a bit faster than WSL2!

**Comment by soyarsauce** [comment]
@nf-s to open pr w/ @zoran995 's fix

**Comment by nf-s** [comment]
Closed by #4742
